test/e2e/autoscale.go

```diff
@@ -61,7 +63,7 @@ const (
 type TestContext struct {
 	t                 *testing.T
 	clients           *test.Clients
-	names             test.ResourceNames
+	names             *test.ResourceNames
```

Comment:
I noticed that other members of this struct are all passed by pointers. Aside from that, on the consumer side we mostly want the pointer instead of the real object. Then it's possible to call directly `test.EnsureTearDown(c.T, ctx.Clients(), ctx.Names())` instead of 
```
names := ctx.Names()
test.EnsureTearDown(c.T, ctx.Clients(), &names)
```